### PR TITLE
feat: migrate to `@wagmi/chains`

### PR DESCRIPTION
**Pull request by jxom**
## Description

This PR migrates wagmi to implement `@wagmi/chains`. Check out the [changesets](https://github.com/wagmi-dev/wagmi/pull/1408/files#diff-7b511dffba09150108f13b455f5a842c15c66d69a00f433f968e3400e3cb4d5aR1).

The `@wagmi/chains` package contains a list of EVM-compatible chains (current chains like Mainnet, Polygon, Optimism, etc, as well as Avalanche, Fantom & BSC), the repository will be public soon!

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/wagmi/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: jxom.eth

### CODE REVIEW

File: .changeset/eighty-insects-tickle.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
+---
+'wagmi': minor
+---
+
+**Breaking**: With the introduction of the [`wagmi/chains` entrypoint](/react/chains#wagmichains), `wagmi` no longer exports the following:
```
Comment (tmm) [inline .changeset/eighty-insects-tickle.md:5]: Technically the `wagmi/chains` entrypoint has already been around, but nbd to spotlight it here.

File: .changeset/fast-points-occur.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+The `multicall` and `ens` attributes have been moved into the `contracts` object:
+
+```diff
+type Chain = {
+  ...
+- multicall: Contract
+- ens: Contract
++ contracts: {
++   multicall3: Contract
++   ensRegistry: Contract
```
Comment (tmm) [inline .changeset/fast-points-occur.md:44]: Think it's useful to show the `Contract` type here too?

File: packages/core/src/chains.ts [hunk 3]
```diff
@@ -1,14 +1,22 @@
...
-  sepolia,
   localhost,
   mainnet,
   optimism,
   optimismGoerli,
   polygon,
   polygonMumbai,
-} from './constants/chains'
+  sepolia,
+} from '@wagmi/chains'
```
Comment (tmm) [inline packages/core/src/chains.ts:20]: Curious what you think about exporting everything from `@wagmi/chains` by default? Might make it easier to bump new versions and we will have the snapshot test as a way to review changes.

File: tsconfig.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
     "noUnusedLocals": true,
     "noUnusedParameters": true,
     "paths": {
+      "@wagmi/chains": ["node_modules/@wagmi/chains/dist"],
```
Comment (tmm) [inline tsconfig.json:18]: would it work if this path was for `@wagmi/chains` in `packages/core/node_modules`? that way we don't need to install `@wagmi/chains` again in the workspace root.

File: docs/package.json [hunk 5]
```diff
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "@reach/skip-nav": "^0.16.0",
+    "@wagmi/chains": "0.0.0-20221201004721",
```
Comment (tmm) [inline docs/package.json:13]: can we remove this from here?

File: docs/pages/core/actions/connect.en-US.mdx [hunk 6]
```diff
@@ -42,16 +42,19 @@ const result = await connect({
...
 import { InjectedConnector } from '@wagmi/core/connectors/injected'
 
 const result = await connect({
-  chainId: chain.optimism.id,
+  chainId: optimism.id,
   connector: new InjectedConnector(),
 })
 ```
 
+Note: The above example is using the `optimism` chain from [`@wagmi/core/chains`](/core/chains#wagmicorechains).
```
Comment (tmm) [inline docs/pages/core/actions/connect.en-US.mdx:56]: Should we make this an `type="info"` `Callout`?

**Review by tmm — COMMENTED** [review]
Dropping some comments. Will finish taking a full look tomorrow!

File: .changeset/fast-points-occur.md (hunk 2 above)
Comment (jxom) [reply .changeset/fast-points-occur.md:44]: Good idea!

File: packages/core/src/chains.ts (hunk 3 above)
Comment (jxom) [reply packages/core/src/chains.ts:20]: Yeah, makes sense to me!

File: docs/pages/core/actions/connect.en-US.mdx (hunk 6 above)
Comment (jxom) [reply docs/pages/core/actions/connect.en-US.mdx:56]: Can change! Thought it might have been too "in your face" though.

Comment (tmm) [reply docs/pages/core/actions/connect.en-US.mdx:56]: Fair! We can keep this as is then.

**Review by tmm — APPROVED** [review]

**Pull request merged**